src/Process/3dTilesProcessing.js

```diff
@@ -88,6 +92,49 @@ export function pre3dTilesUpdate(context, layer) {
     // const HFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) / context.camera.ratio);
     const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
     context.camera.preSSE = hypotenuse * (2.0 * Math.tan(HYFOV * 0.5));
+
+    // once in a while, garbage collect
+    if (Math.random() < 0.98) {
```

Comment:
Is there a reason for using a random test instead of a fixed waiting time?

Reply:
Nope, the random() solution allows to not maintain a state (`lastCleanupTime` or similar).

Another option would be to maintain a `cleanable` array with all potentially removable objects are referenced.
This array could be browsable at each update loop easily.